lib/Doctrine/ORM/Mapping/AssociationMapping.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+use Exception;
+
+use function assert;
+use function in_array;
+use function property_exists;
+
+/** @template-implements ArrayAccess<string, mixed> */
+abstract class AssociationMapping implements ArrayAccess
```

Comment:
Can we `final`ize all public methods in this class that are not overridden by child classes?